src/controllers/UserController.ts

```diff
@@ -1,52 +1,75 @@
     }
 
-    public getAutoSuggestUsers = async (req: Request, res: Response): Promise<void> => {
+    public getAutoSuggestUsers = async (req: Request, res: Response, next: (arg: any) => void): Promise<void> => {
         try {
-            res.send(await this.userService.getUsers(req.query.login, req.query.limit));
+            const users: Array<UserType> = await this.userService.getUsers(req.query.login, req.query.limit);
+
+            if (users.length === 0) {
+                throw new Api404Error('Users not found.');
+            }
+
+            res.send(users);
         } catch (error) {
-            res.status(404).send(error);
+            return next(error);
         }
     }
 
-    public getUserById = async (req: Request, res: Response): Promise<void> => {
+    public getUserById = async (req: Request, res: Response, next: (arg: any) => void): Promise<void> => {
         try {
-            await this.userService.handleIdError(req.params.id);
-            res.json(await this.userService.getUserById(req.params.id));
+            await this.handleErrorNotFoundByUserId(req.params.id as unknown as number);
+
+            res.json(await this.userService.getUserById(req.params.id as unknown as number));
         } catch (error) {
-            res.status(404).json({ message: 'id not found' });
+            return next(error);
         }
     }
 
     public createUser = async (req: Request, res: Response): Promise<void> => {
         try {
-            res.send(await this.userService.createUser(req.query));
+            const user: UserType = await this.userService.createUser(req.query);
+            res.status(201).send(await this.userService.createUser(user));
         } catch (error) {
-            res.status(404).send(error);
+            logError(error);
         }
     }
 
-    public updateUserById = async (req: Request, res: Response): Promise<void> => {
+    public updateUserById = async (req: Request, res: Response, next: (arg: any) => void): Promise<void> => {
         try {
-            await this.userService.handleIdError(req.params.id);
+            await this.handleErrorNotFoundByUserId(req.params.id as unknown as number);
+
             res.send(await this.userService.updateUserById(req.query, req.params.id));
         } catch (error) {
-            res.status(404).json({ message: 'id not found' });
+            return next(error);
         }
     }
 
-    public removeUserById = async (req: Request, res: Response): Promise<void> => {
+    public removeUserById = async (req: Request, res: Response, next: (arg: any) => void): Promise<void> => {
         try {
-            await this.userService.handleIdError(req.params.id);
-            res.send(await this.userService.removeUserById(req.params.id));
+            await this.handleErrorNotFoundByUserId(req.params.id as unknown as number);
+
+            res
+                .json({ message: 'User has been deleted.' })
```

Comment:
It seems that you have an error here. Calling send method after json method will throw an error "headers already sent". Json method is same with send method, with one difference - json add header Content-Type